### DISCUSSION TITLE
Using 2D and 3D views with CompositeViewer

### DISCUSSION THREAD

**Discussion by keince** [pelicanmapping/osgearth#1831] [Support Q&A]
After doing some prototyping with using multiple views with the CompositeViewer, we've decided to expand our application to support the multiview paradigm. However, I do have some questions moving forward...

We would like to display a 2D projected environment (spherical mercator) in one view and a 3D geocentric (wgs84) of the same environment in another view.

One obvious question is, can I share the same scenegraph or map node or do this or do I need to use two different scenegraphs or map nodes to do this efficiently? One for the 2D projected view and another for the 3D view? It appears that I am able to display wgs84 maps with a map type of "spherical mercator" but it is very slow loading probably due to the fact that it must reproject the maps during runtime.

If I must use two different map nodes for the projected 2D and geocentric 3D views, can I define two different map nodes in an earth file or do I need to use two different earth files?

If someone can give me feedback on the best, most efficient way forward in accomplishing the task, I'd appreciate it...

Shayne

**Comment by keince** [comment]
It appears that each map must be specified in its own earth file so that question has been answered...

**Reply by gwaldron** [reply]
Yes, that's right. The map projection is tied to the Map itself in osgEarth, so you cannot have one map that's rendered in multiple different projections.

The fastest 2D projection is "plate-carre" which treats lat=y and long=x directly, so no reprojection is necessary. Mercator is usually a pretty fast reprojection though.

**Reply by keince** [reply]
Thank you for the feedback. 

So, it sounds like the best approach is to use two different maps, one for each view? 

There isn't a way to share one map between them and reproject the map layer for the flat earth view?

**Reply by gwaldron** [reply]
Correct on both counts.

**Reply by keince** [reply]
One more question...

Can subgraphs be shared between mapNodes? For example, I have a list of entityNode (subclassed from TrackNode) in a osg::Group called entityGroup. Can I share this entityGroup with two different mapNodes or scenegraphs? My entityNode class supports both projected and geocentric.

Here are my scenegraphs showing what I'd like to do...sharing the entityGroup with two different scenegraphs, one projected, the other geocentric:

**2D scenegraph:**

root2D
    |
mapNode2D
    |
**entityGroup**

**3D scenegraph:**

root3D
    |
mapNode3D
    |
**entityGroup**
